### PR TITLE
add starter of custom 404/422/500 routing

### LINKED ISSUE #2474: Better error pages
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

Could use something a little snappier than the rails default 

image

* **What feature or behavior is this required for?**

* **How could we solve this issue? (Not knowing is okay!)**

* **Anything else?**

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Roughly following  https://westonganger.com/posts/creating-custom-error-pages-with-rails

This pull request makes the following changes:
* removes stock rails error pages
* adds custom error pages based on @emtot22 's mockups (thank you)

here's what they look like in practice, logged in and not:

image

image

You can try em out at localhost:3000/400, 422, 500

It relates to the following issue #s: 
* Fixes #2474 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -3,32 +3,46 @@
   <head>
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
     <title><%= content_for?(:title) ? yield(:title) : "DARIA" %></title>
-    <meta name="description" content="<%= content_for?(:description) ? yield(:description) : "DCAF" %>">
+    <meta name="description" content="<%= content_for?(:description) ? yield(:description) : "DARIA" %>">
     <meta name='ROBOTS' content='NOINDEX, NOFOLLOW'>
     <meta name="google-site-verification" content="<%= ENV['GOOGLE_SITE_VERIFICATION'] %>" />
 
-    <%= csrf_meta_tags %>
+    <% if !content_for? :error_page %>
```
Comment (colinxfleming) [inline app/views/layouts/application.html.erb:10]: With some gentle control flow here we can reuse the main layout. Essentially, the reasoning is: If there was an error, no need to set up CSRF, render async, or flash messages.

File: config/environments/development.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@
   config.eager_load = false
 
   # Show full error reports.
-  config.consider_all_requests_local = true
+  config.consider_all_requests_local = false
```
Comment (colinxfleming) [inline config/environments/development.rb:15]: fine for development mode.

File: config/routes.rb [hunk 3]
```diff
@@ -64,4 +64,8 @@
     get '/users/edit' => 'devise/registrations#edit', as: 'edit_user_registration'
     put '/users' => 'devise/registrations#update', as: 'registration'
   end
+
+  match '/404' => 'errors#error_404', via: [:get, :post, :put, :patch]
```
Comment (colinxfleming) [inline config/routes.rb:68]: You can see these pages for yourself by hitting localhost:3000/400, localhost:3000/422, and localhost:3000/500

**Review by colinxfleming — COMMENTED** [review]
couple notes

**Comment by emtot22** [comment]
Looks awesome!!

File: app/views/errors/error_500.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<% content_for :title do %>DARIA - 500<% end %>
+<% content_for :error_page do %>error<% end %>
+
+<h2>Sorry! Something went wrong. (500)</h2>
+
+<p>This is probably an us problem - try again later, and let us know if it persists.</p>
```
Comment (lomky) [inline app/views/errors/error_500.html.erb:6]: Might be worth specifying who 'us' is here (Daria maintainers vs the Fund). Maybe:

```suggestion
<p>This is probably an us problem - try again later, and have your organization let the Daria maintainers know if it persists.</p>
```

**Review by lomky — APPROVED** [review]
🎉

**Pull request merged**